upgrade/helpers/tasks.py

```diff
@@ -923,3 +923,32 @@ def repository_cleanup(repo_name):
     for fname in os.listdir('/etc/yum.repos.d/'):
         if repo_name in fname.lower():
             os.remove('/etc/yum.repos.d/{}'.format(fname))
+
+
+def update_scap_content():
+    """ The purpose of this function is to perform deletion of old scap-contents
+    and then uploading new scap-contents. It also updates scap-policies and assign
+     new scap-content to them """
+    sc = entities.ScapContents().search()
+    sc_ids = []
+    cp = entities.CompliancePolicies().search()
+    cp_ids = []
+    for i in range(len(sc)):
+        tmp_sc_id = sc[i].id if sc[i].title == "Red Hat firefox default content" else 1
+        sc_ids.append(sc[i].id) if sc[i].title in scap_content else sc_ids
+    for i in range(len(cp)):
+        if cp[i].scap_content_id in sc_ids:
+            cp_ids.append(cp[i].id)
+            entities.CompliancePolicies(
+                id=cp[i].id, scap_content_id=tmp_sc_id,
+                scap_content_profile_id=tmp_sc_id
+            ).update()
+    for i in range(len(sc_ids)):
+        entities.ScapContents(id=sc_ids[i]).delete()
+    run('foreman-rake foreman_openscap:bulk_upload:default')
+    for i in range(len(cp)):
```

Comment:
Can we create a common function to reduce the repeated look kind of things?

Reply:
I am trying to say like below.
def update_scap_content():
    def scap_content(content_type, content_name):
        for _ in range(len(content_type)):
            if content_name == "sc":
                pass
            elif content_name == "cp":
                pass
            elif content_name == "sc_ids":
                pass
            elif content_name == "cp_id":
                pass
    sc = entities.ScapContents().search()
    sc_ids = []
    cp = entities.CompliancePolicies().search()
    cp_ids = []
    scap_content(sc, "sc")
    scap_content(cp, "cp")
    scap_content(sc_ids, "sc_ids")
    run('foreman-rake foreman_openscap:bulk_upload:default')
    scap_content(cp, "cp_id")

Reply:
Thanks @devendra104 
I have updated the PR.